source/contribute.rst

```diff
@@ -270,73 +311,20 @@ with both the code and the docs here.
 Rebasing on upstream changes
 ----------------------------
 
-Next, merge or rebase your commit against **doctrine/2.11.x**. With your
-work done in a **local** topic branch, you'll want to assist upstream
-merge by rebasing your commits. You can either do this manually with
-``fetch`` then ``rebase``, or use the ``pull --rebase`` shortcut. You
-may encounter merge conflicts, which you should fix and then mark as
-fixed with ``add``, and then continue rebasing with
-``rebase --continue``. At any stage, you can abort the rebase with
-``rebase --abort`` unlike nasty merges which will leave files strewn
-everywhere.
-
-.. code-block:: console
-
-    $ git fetch doctrine
-    $ git rebase doctrine/2.11.x fix-weird-bug
-
-Push your branch to **origin**:
-
-Finished topic branches should be pushed to **origin** for a
-**maintainer** to review and pull into **doctrine** as appropriate:
-
-.. code-block:: console
-
-    $ git push origin fix-weird-bug
-    To git@github.com:hobodave/dbal.git
-        * [new branch]      fix-weird-bug -> fix-weird-bug
-
-Now you are ready to send a pull request from this branch and ask for a
-review from a maintainer.
-
-Topic Branch Cleanup
---------------------
-
-Once your work has been merged by the branch maintainer, it will no
-longer be necessary to keep the local branch or remote branch, so you
-can remove them!
-
-Sync your local 2.11.x branch:
-
-.. code-block:: console
+Sometimes, you will need to rebase your branch on the latest changes,
+typically because the build had an issue unrelated to your changes, and
+that issue has been fixed after you created your branch.
 
-    $ git checkout 2.11.x
-    $ git pull --rebase
-
-Remove your local topic branch using -d to ensure that it has been merged by
-upstream. Branch -d will not delete a branch that is not an ancestor of
-your current head.
-
-From the git-branch man page:
-
-.. code-block:: console
-
-    -d
-        Delete a branch. The branch must be fully merged in HEAD.
-    -D
-        Delete a branch irrespective of its merged status.
-
-Remove your local topic branch:
-
-.. code-block:: console
-
-    $ git branch -d fix-weird-bug
-
-Remove your remote branch at **origin**:
-
-.. code-block:: console
+Here is how to proceed if you need to rebase on ``2.3.x``:
 
-    $ git push origin :fix-weird-bug
+1. Fetch all new commits: `git fetch doctrine``.
+2. Rebase on what you fetched interactively:
+   ``git rebase --interactive doctrine/2.3.x # use -i for short``
+3. A window will show up with many lines, remove every line but the last
+   ones, which should correspond to your commits.
```

Comment:
Hm, that's pretty vague, and every line removed potentially is a lost commit. A nice footgun, eh?

As long as you're only rebasing on the same branch that you initially started out with, does it have to be an interactive rebase at all?

Reply:
(Assuming we're writing all this for people rather new to git and contributing on GitHub, so we should not talk them into complicated situations where they might lose work and/or get frustrated)

Reply:
Cool

Reply:
Let’s have a dedicated section about changing the target branch and the rebase operations that are necessary for this. 

Additionally, prepare a canned reply that we can use in PRs whenever we ask people to change the target branch. This should point to that documentation, mention the correct branch and tell people that they _need not_ close the PR but can update it.

Reply:
Addressed.